MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -185,10 +202,26 @@ - (void)deleteLogsFromDBWithColumnValues:(NSArray<NSString *> *)columnValues col
 
   // Execute.
   if ([self.connection executeQuery:deleteLogsQuery]) {
-    MSLogVerbose([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"succeded");
+    MSLogVerbose([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"succeeded");
   } else {
     MSLogError([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"failed");
   }
 }
 
+- (void)deleteOldestLogWithGroupId:(NSString *)groupId {
+  NSString *deleteLogQuery =
+      [NSString stringWithFormat:@"DELETE FROM %@ WHERE %@ = '%@' ORDER BY %@ ASC LIMIT 1", kMSLogTableName,
```

Comment:
There is a high change of having logs more than limit since you are adding and then deleting log when it exceeds the limit. If 301th log gets added but it isn't able to delete the first log somehow, next time it will maintain 301 logs. Why don't you delete multiple logs that are exceeding the limit?